samplers/aws/xray/sampling_rule.go

```diff
@@ -77,3 +82,122 @@ func (r *centralizedRule) updateRule(rule *ruleProperties) {
 	r.ruleProperties = rule
 	r.reservoir.capacity = *rule.ReservoirSize
 }
+
+// Sample returns SamplingResult with SamplingDecision, TraceState and Attributes
+func (r *centralizedRule) Sample(parameters sdktrace.SamplingParameters) sdktrace.SamplingResult {
+	attributes := []attribute.KeyValue{
+		attribute.String("Rule", *r.ruleProperties.RuleName),
+	}
+
+	sd := sdktrace.SamplingResult{
+		Attributes: attributes,
+		Tracestate: trace.SpanContextFromContext(parameters.ParentContext).TraceState(),
+	}
+
+	now := r.clock.Now().Unix()
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	r.matchedRequests++
+
+	// fallback sampling logic if quota has expired
+	if r.reservoir.expired(now) {
+		// if reservoir quota is expired then sampling using
+		// sampling config: 1 req/sec and then x% of fixedRate
+
+		// Sampling 1 req/sec
+		if r.reservoir.borrow(now) {
+			globalLogger.Printf(
+				"Sampling target has expired for rule %s. Using fallback sampling and borrowing 1 req/sec from reservoir",
+				*r.ruleProperties.RuleName,
+			)
+			r.borrowedRequests++
+
+			sd.Decision = sdktrace.RecordAndSample
+			return sd
+		}
+
+		globalLogger.Printf(
+			"Sampling target has expired for rule %s. Using traceIDRationBased sampler to sample 5 percent of requests during that second",
+			*r.ruleProperties.RuleName,
+		)
+
+		// using traceIDRatioBased sampler to sample
+		samplingDecision := sdktrace.TraceIDRatioBased(*r.ruleProperties.FixedRate).ShouldSample(parameters)
+
+		samplingDecision.Attributes = attributes
+
+		if samplingDecision.Decision == sdktrace.RecordAndSample {
+			r.sampledRequests++
+		}
+
+		return samplingDecision
+	}
+
+	// Take from reservoir quota, if possible
+	if r.reservoir.Take(now) {
+		r.sampledRequests++
+		sd.Decision = sdktrace.RecordAndSample
+
+		return sd
+	}
+
+	globalLogger.Printf(
+		"Sampling target has been exhausted for rule %s. Using traceIDRatioBased Sampler with fixed rate.",
+		*r.ruleProperties.RuleName,
+	)
+
+	// using traceIDRatioBased sampler to sample using fixed rate
+	samplingDecision := sdktrace.TraceIDRatioBased(*r.ruleProperties.FixedRate).ShouldSample(parameters)
+
+	samplingDecision.Attributes = attributes
+	samplingDecision.Tracestate = trace.SpanContextFromContext(parameters.ParentContext).TraceState()
```

Comment:
```suggestion
	samplingDecision.Tracestate = sd.Tracestate
```

THIs has already been calculated once, no need to do it again.